libs/vm/include/vm/array.hpp

```diff
@@ -179,6 +182,17 @@ struct Array : public IArray
     std::reverse(elements.begin(), elements.end());
   }
 
+  void Extend(Ptr<IArray> const &other) override
+  {
+    Ptr<Array<ElementType>> const &other_array    = other;
+    auto const &                   other_elements = other_array->elements;
+
+    elements.reserve(elements.size() + other_elements.size());
+
+    auto insert_iterator = std::back_inserter(elements);
+    std::copy(other_elements.cbegin(), other_elements.cend(), insert_iterator);
```

Comment:
Why not `elements.insert()`?

Reply:
Good idea. I'd rather leave the `reserve` call in, though, as cppreference mentions no guarantees that `insert` reserves the required space in advance.

Reply:
Well, it mentions that time complexity is linear which implies at-most O(1) reallocations, which in turn implies pre-reservation, but ok.